### PR TITLE
#170947437 Build password reset feature

### PR DESCRIPTION
**Pull request by RusimbiPatrick**
### What does this PR do?

- Create Forgot password endpoint
- Create reset password endpoint
- Remove useless comments
- Write tests for reset password feature
- add validation for newPassword field

### How should this be manually tested?

- Clone the [repo](https://github.com/andela/octopus-bn-backend.git)
- Run ``` npm install ```
-  Hit 
  -  `POST ``http://localhost:3000/api/v1/auth/forgot-password` to request a password reset link.
  - Check email for the confirmation link and run it using a `POST` method

### Implementation plan

[Implementation plan name](https://docs.google.com/document/d/11UF7-_F-_vS5aZ0oAmVbuYwDeEZfZS_sHjq6U-h0vXQ/edit)

### Stories associated with PR

[##170947435](https://www.pivotaltracker.com/story/show/170947437)

### Screenshots if any
Screen Shot 2020-02-18 at 20 11 16
Screen Shot 2020-02-18 at 20 11 53

### CODE REVIEW

File: locales/en.js [hunk 1]
```diff
@@ -11,5 +11,9 @@
...
 	"localLoginSuccess": "Successfully logged in",
 	"internalServerError": "This service is currently unavailable. Developers at barefoot nomad are maintaining it",
-	"outSuccessfully": "Logout successful"
-}
+	"outSuccessfully": "Logout successful",
+	"emailSent": "We have sent you an email",
+	"passwordReset": "Password reset succesfully",
+	"password": "password",
+	"passwordsDontMatch": "Passwords do not match"
+}
```
Comment (blaise82) [inline locales/en.js]:
```suggestion
}

```
please add a new line

File: src/routes/api/user.routes.js [hunk 2]
```diff
@@ -221,4 +222,72 @@ router.post('/signin', checkValidEmail, checkPassword, userController.signin);
...
+ *             properties:
+ *               status:
+ *                 type: integer
+ *               message:
+ *                 type: string
+ *     responses:
+ *       200:
+ *         description: We have sent you an email
+ *  */
+router.post('/forgot-password', checkValidEmail, validateResult, userController.forgotPassword);
```
Comment (blaise82) [inline src/routes/api/user.routes.js:411]: forget passport is more of a put than a post

Comment (blaise82) [inline src/routes/api/user.routes.js:411]:
```suggestion
router.put('/forgot-password', checkValidEmail, validateResult, userController.forgotPassword);
```

**Review by hakoemmy — APPROVED** [review]

File: src/routes/api/user.routes.js (hunk 2 above)
Comment (RusimbiPatrick) [reply src/routes/api/user.routes.js:411]: Thank you @blaise82. I put `PUT` on the reset password route instead. Because it is where I am updating from

**Review by blaise82 — APPROVED** [review]

**Comment by AbdoulNuru** [comment]
@RusimbiP The reset password link sent in the email is not working? I even tested using the route on the documentation but it still gives a not found error.

**Comment by RusimbiPatrick** [comment]
I am currently rebasing

**Review by AbdoulNuru — APPROVED** [review]

**Review by kcharles52 — APPROVED** [review]

**Pull request merged**